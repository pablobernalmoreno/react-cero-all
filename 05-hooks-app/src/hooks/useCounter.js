import { useState } from "react";

export const useCounter = (intialState = 0) => {
  const [state, setState] = useState(intialState);

  const increment = (factor = 1) => {
    setState(state + factor);
  };

  const decrement = (factor = 1) => {
    setState(state - factor);
  };

  const reset = () => {
    setState(intialState);
  };
  return {
    state,
    increment,
    decrement,
    reset,
  };
};
