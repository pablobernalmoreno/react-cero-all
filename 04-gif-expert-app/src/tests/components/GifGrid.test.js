import React from "react";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en GifGrid", () => {
  const gifs = [
    {
      id: "id",
      url: "https://",
      title: "title",
    },
  ];

  useFetchGifs.mockReturnValue({
    data: [],
    loading: true,
  });

  const category = "One Piece";
  const wrapper = shallow(<GifGrid category={category} />);

  test("debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar items cuando se cargan las imagenes con useFetchGifs", () => {
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
