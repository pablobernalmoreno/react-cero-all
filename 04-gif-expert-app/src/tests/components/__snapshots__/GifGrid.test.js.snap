// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3>
    One Piece
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
